azure-kusto-data/tests/test_kusto_client.py

```diff
@@ -341,43 +327,186 @@ def test_admin_then_query(self, mock_post):
+        """Test query V2."""
+        client = self._create_kusto_client()
+        response = client.execute_query("PythonTest", "Deft")
+        self._assert_sanity_query_response(response)
+
+    @patch("requests.Session.post", side_effect=mocked_requests_post)
+    def test_sanity_control_command(self, mock_post):
+        """Tests contol command."""
+        client = self._create_kusto_client()
+        response = client.execute_mgmt("NetDefaultDB", ".show version")
+        self._assert_sanity_control_command_response(response)
+
+    @pytest.mark.skipif(not PANDAS, reason="requires pandas")
+    @patch("requests.Session.post", side_effect=mocked_requests_post)
+    def test_sanity_data_frame(self, mock_post):
+        """Tests KustoResponse to pandas.DataFrame."""
+        client = self._create_kusto_client()
+        response = client.execute_query("PythonTest", "Deft")
+        data_frame = dataframe_from_result_table(response.primary_results[0])
+        self._assert_sanity_data_frame_response(data_frame)
+
+    @patch("requests.Session.post", side_effect=mocked_requests_post)
+    def test_partial_results(self, mock_post):
+        """Tests partial results."""
+        client = self._create_kusto_client()
+        query = """set truncationmaxrecords = 5;
+range x from 1 to 10 step 1"""
+        properties = ClientRequestProperties()
+        properties.set_option(ClientRequestProperties.results_defer_partial_query_failures_option_name, False)
+        self.assertRaises(KustoServiceError, client.execute_query, "PythonTest", query, properties)
+        properties.set_option(ClientRequestProperties.results_defer_partial_query_failures_option_name, True)
+        response = client.execute_query("PythonTest", query, properties)
+        self._assert_partial_results_response(response)
+
+    @patch("requests.Session.post", side_effect=mocked_requests_post)
+    def test_admin_then_query(self, mock_post):
+        """Tests admin then query."""
+        client = self._create_kusto_client()
+        query = ".show tables | project DatabaseName, TableName"
+        response = client.execute_mgmt("PythonTest", query)
+        self._assert_admin_then_query_response(response)
+
+    @patch("requests.Session.post", side_effect=mocked_requests_post)
+    def test_dynamic(self, mock_post):
+        """Tests dynamic responses."""
+        client = self._create_kusto_client()
+        query = """print dynamic(123), dynamic("123"), dynamic("test bad json"),"""
+        """ dynamic(null), dynamic('{"rowId":2,"arr":[0,2]}'), dynamic({"rowId":2,"arr":[0,2]})"""
+        row = client.execute_query("PythonTest", query).primary_results[0].rows[0]
+        self._assert_dynamic_response(row)
+
     @patch("requests.Session.post", side_effect=mocked_requests_post)
     def test_empty_result(self, mock_post):
         """Tests dynamic responses."""
-        client = KustoClient("https://somecluster.kusto.windows.net")
+        client = self._create_kusto_client()
```

Comment:
revert. I believe in explicit creation over implicit, this code hides the actual value for no real reason.